nilearn/conftest.py

```diff
@@ -377,3 +371,167 @@ def img_atlas(shape_3d_default, affine_mni):
+    9 vertices and 10 faces.
+
+    The left part is a tetrahedron with four vertices and four faces.
+    The right part is a pyramid with five vertices and six faces.
+    """
+    left_coords = np.asarray([[0.0, 0, 0], [1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    left_faces = np.asarray([[1, 0, 2], [0, 1, 3], [0, 3, 2], [1, 2, 3]])
+    right_coords = (
+        np.asarray([[0.0, 0, 0], [1, 0, 0], [1, 1, 0], [0, 1, 0], [0, 0, 1]])
+        + 2.0
+    )
+    right_faces = np.asarray(
+        [
+            [0, 1, 4],
+            [0, 3, 1],
+            [1, 3, 2],
+            [1, 2, 4],
+            [2, 3, 4],
+            [0, 4, 3],
+        ]
+    )
+    return PolyMesh(
+        left=InMemoryMesh(left_coords, left_faces),
+        right=InMemoryMesh(right_coords, right_faces),
+    )
+
+
+@pytest.fixture()
+def surf_mesh():
+    """Return _make_mesh as a function allowing it to be used as a fixture."""
+    return _make_mesh
+
+
+@pytest.fixture
+def surf_img():
+    """Create a sample surface image using the sample mesh.
+    This will add some random data to the vertices of the mesh.
+    The shape of the data will be (n_samples, n_vertices).
+    n_samples by default is 1.
+    """
+
+    def _make_surface_img(n_samples=(1,)):
+        mesh = _make_mesh()
+        data = {}
+        for i, (key, val) in enumerate(mesh.parts.items()):
+            data_shape = (*tuple(n_samples), val.n_vertices)
+            data_part = (
+                np.arange(np.prod(data_shape)).reshape(data_shape) + 1.0
+            ) * 10**i
+            data[key] = data_part
+        return SurfaceImage(mesh, data)
+
+    return _make_surface_img
+
+
+@pytest.fixture
+def surf_mask():
+    """Create a sample surface mask using the sample mesh.
+    This will create a mask with n_zeros zeros (default is 4) and the
+    rest ones. If empty is True, the mask will be None, required for
+    tests for html reports.
+    """
+
+    def _make_surface_mask(n_zeros=4, empty=False):
+        if empty:
+            return None
+        else:
+            mesh = _make_mesh()
+            data = {}
+            for key, val in mesh.parts.items():
+                data_part = np.ones(val.n_vertices, dtype=int)
+                for i in range(n_zeros // 2):
+                    data_part[..., i] = 0
+                data_part = data_part.astype(bool)
+                data[key] = data_part
+            return SurfaceImage(mesh, data)
+
+    return _make_surface_mask
+
+
+@pytest.fixture
+def surf_label_img():
+    """Return a sample surface label image using the sample mesh.
+    Has two regions with values 0 and 1 respectively.
+    """
+
+    def _surface_label_img():
+        mesh = _make_mesh()
+        data = {
+            "left": np.asarray([0, 0, 1, 1]),
+            "right": np.asarray([1, 1, 0, 0, 0]),
+        }
+
+        return SurfaceImage(mesh, data)
+
+    return _surface_label_img
+
+
+@pytest.fixture
+def flip_surf_img_parts():
```

Comment:
It still feels weird to me to use the last four as fixture and not as functions that we store somewhere in nilearn._utils ? Maybe nilearn._utils.testing?

But I don't think this is urgent. Let's open a side issue for these and move on to merge.

Reply:
probably something to do as we try to also better fixturize test for the old surface code